core/document/src/document.rs

```diff
@@ -226,6 +226,15 @@ impl Document {
 		Ok(())
 	}
 
+	pub fn reorder_layers(&mut self, source_paths: &[Vec<LayerId>], target_path: &[LayerId]) -> Result<(), DocumentError> {
+		// TODO: Detect when moving between folders and handle properly
+
+		let source_layer_ids = source_paths.iter().map(|x| *x.last().unwrap()).collect();
```

Comment:
You could throw document Errors here instead of just panicking because this is a public function